src/main/java/org/openjdk/jextract/impl/StructBuilder.java

```diff
@@ -282,70 +285,71 @@ private void emitFieldArraySetter(String javaName, Declaration.Variable varTree,
         appendBlankLine();
         emitFieldDocComment(varTree, "Indexed setter for field:");
         if (Utils.isStructOrUnion(elemType)) {
-            appendIndentedLines(STR."""
-                public static void \{javaName}(MemorySegment \{segmentParam}, \{indexList.decl()}, MemorySegment \{valueParam}) {
-                    MemorySegment.copy(\{valueParam}, 0L, \{javaName}(\{segmentParam}, \{indexList.use()}), 0L, \{layoutString(elemType)}.byteSize());
+            appendIndentedLines("""
+                public static void %1$s(MemorySegment %2$s, %3$s, MemorySegment %4$s) {
+                    MemorySegment.copy(%4$s, 0L, %1$s(%2$s, %5$s), 0L, %6$s.byteSize());
                 }
-                """);
+                """, javaName, segmentParam, indexList.decl()
+                   ,valueParam, indexList.use(), layoutString(elemType));
         } else {
-            appendIndentedLines(STR."""
-                public static void \{javaName}(MemorySegment \{segmentParam}, \{indexList.decl()}, \{elemTypeCls.getSimpleName()} \{valueParam}) {
-                    \{arrayElementHandle}.set(\{segmentParam}, 0L, \{indexList.use()}, \{valueParam});
+            appendIndentedLines("""
+                public static void %1$s(MemorySegment %2$s, %3$s, %4$s %5$s) {
+                    %6$s.set(%2$s, 0L, %7$s, %5$s);
                 }
-                """);
+                """, javaName, segmentParam, indexList.decl(), elemTypeCls.getSimpleName()
+                   ,valueParam, arrayElementHandle, indexList.use());
         }
     }
 
     private void emitAsSlice() {
         String arrayParam = safeParameterName("array");
-        appendIndentedLines(STR."""
+        appendIndentedLines(MessageFormat.format("""
 
             /**
-             * Obtains a slice of {@code arrayParam} which selects the array element at {@code index}.
-             * The returned segment has address {@code arrayParam.address() + index * layout().byteSize()}
+             * Obtains a slice of {0} which selects the array element at {0}.
+             * The returned segment has address {0}.address() + index * layout().byteSize()
              */
-            public static MemorySegment asSlice(MemorySegment \{arrayParam}, long index) {
-                return \{arrayParam}.asSlice(layout().byteSize() * index);
-            }
-            """);
+            public static MemorySegment asSlice(MemorySegment {0}, long index) '{'
+                return {0}.asSlice(layout().byteSize() * index);
+            '}'
+            """, arrayParam));
```

Comment:
This doesn't look right. The javadoc should be changed here.

Reply:
The quoting of the `{` also seems suspicious?

Reply:
This was one of the couple of times where I used `MessageFormat`, so I had to escape the `{`

Reply:
This was one of the couple of times where I used MessageFormat, so I had to escape the {

Reply:
I presume we can just drop the call to `Message::format` here and keep `appendIndentedLines`, right?

Reply:
Sure, I will do that.
I will try to cover all the suggested fixes in one patch